### PR TITLE
Note deletion confirmation

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
This PR prompts admins to confirm they really do want to delete a note completely.

## Why It's Good For The Game
Deleting notes outright is not something done lightly, the fact that you can misclick and do it without confirmation is... troublesome. This adds some sort of safety net to doing so.

## Images of changes
2021-09-10 23_08_17-Note deletion confirmation

## Changelog
:cl:
tweak: Admins now have to confirm they want to delete a note.
/:cl:

### CODE REVIEW

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -982,7 +982,9 @@
 
 	else if(href_list["removenote"])
 		var/note_id = href_list["removenote"]
-		remove_note(note_id)
+		if(note_id)
+			if((alert("Do you really want to delete this note?","Note deletion confirmation", "Yes", "No") == "Yes") && note_id)
+				remove_note(note_id)
```
Comment (AffectedArc07) [inline code/modules/admin/topic.dm:987]: Ok you check the note ID on 985, then again on 986, then again inside the proc (sql_notes.dm line 95).


Clean this up.

**Review by AffectedArc07 — DISMISSED** [review]

File: code/modules/admin/topic.dm (hunk 1 above)
Comment (S34NW) [reply code/modules/admin/topic.dm:987]: gotta make sure my note didn't go walkies.
(I'll change it)

**Review by variableundefined — APPROVED** [review]

File: code/modules/admin/topic.dm [hunk 2]
```diff
@@ -982,7 +982,8 @@
 
 	else if(href_list["removenote"])
 		var/note_id = href_list["removenote"]
-		remove_note(note_id)
+		if(alert("Do you really want to delete this note?","Note deletion confirmation", "Yes", "No") == "Yes")
```
Comment (SabreML) [inline code/modules/admin/topic.dm:985]:
```suggestion
		if(alert("Do you really want to delete this note?", "Note deletion confirmation", "Yes", "No") == "Yes")
```

**Review by SabreML — APPROVED** [review]

File: code/modules/admin/topic.dm (hunk 2 above)
Comment (S34NW) [reply code/modules/admin/topic.dm:985]: ah thanks, there is always something to find eh :P

**Review by Kyep — APPROVED** [review]

**Review by variableundefined — APPROVED** [review]

**Pull request merged**